src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -517,13 +518,9 @@ _mongoc_get_rr_search (const char *service,
 
       num_matching_records++;
 
-      if (rr_data) {
```

Comment:
I assume that this check was removed because `rr_data` is already accessed in L489. However, it is not asserted against, so it could be `NULL` after all. Would it make sense to add an assertion at the top to guard against this?

Reply:
Yeah that seems like a good idea. Adding a `BSON_ASSERT (rr_data)`